scripts/test-upgrade.sh

```diff
@@ -106,14 +106,48 @@ ansible galera_all[0] -m shell -a "mysql --verbose -e \"${VLAN_FLAT}\" neutron"
 ansible galera_all[0] -m shell -a "mysql --verbose -e \"${VXLAN}\" neutron"
 
 # TASK #6
+# Bug:   https://github.com/rcbops/u-suk-dev/issues/383
+# Issue: The ceph-all.yml playbook will fail because of the hostname
+#        changes introduced in Mitaka. This section recreates the mons
+#        in a rolling fashion so quorum is maintained, and hostnames
+#        are properly referenced in the ceph datasets(i.e monmap)
+# NOTE:  Before running test-upgrade.sh on an AIO with ceph, variables
+#        in the overrides file will need to be generated manually by
+#        running migrate-yaml.py(similiar to TASK #0). The deployer
+#        must ensure that osd_directory and raw_mutli_journal are not
+#        both set to True in user_rpco_variables_overrides.yml. If this
+#        isn't done, this TASK will fail.
+#        In gating this is handled by gate-specific overrides set in
+#        the jenkins-rpc repository.
+
+# These are the instructions to give if there is a failure message on ceph-mon.yml
+mons=$(ansible mons --list-hosts)
+if [ $(echo ${mons} | wc -w) -gt 0 ]; then
+  #Gather facts about all the monitors first, even with bad names
+  #This shouldn't change a thing, it's a regular playbook run.
+  openstack-ansible ${RPCD_DIR}/playbooks/gen-facts.yml
+  for mon in ${mons}; do
+    ansible $mon -m command -a "stop ceph-mon id=$mon"
+    ansible $mon -m command -a "ceph mon remove $mon"
+    openstack-ansible lxc-containers-destroy.yml --skip-tags=container-directories --limit $mon
+    openstack-ansible lxc-containers-create.yml --limit $mon
+    # Destroy facts for the current mon, this way it's gonna be properly generated
+    rm -f /etc/openstack_deploy/ansible_facts/$mon
+    openstack-ansible ${RPCD_DIR}/playbooks/ceph-mon.yml --limit $mon
```

Comment:
Good question.
It's worth another commit IMO, and if possible I'd verify the health of the cluster without relying on the presence or not of maas.